src/test/components/ListOfPublishedProfiles.test.js

```diff
@@ -443,4 +443,46 @@ describe('ListOfPublishedProfiles', () => {
       );
     });
   });
+
+  describe('The uploaded recordings list should update on window focus', () => {
+    it('will update the stored profiles on window focus', async function() {
+      // Add 3 examples, all with the same name.
+      // mockDate('4 Jul 2020 15:00'); // Now is 4th of July, at 3pm local timezone.
+  
+      const exampleProfileData = {
+        profileToken: 'MACOSX',
+        jwtToken: null,
+        publishedDate: new Date('4 Jul 2020 13:00'),
+        name: 'Example Profile',
+        preset: null,
+        originHostname: 'https://mozilla.org',
+        meta: {
+          product: 'Firefox',
+          platform: 'Macintosh',
+          toolkit: 'cocoa',
+          misc: 'rv:62.0',
+          oscpu: 'Intel Mac OS X 10.12',
+        },
+        urlPath: '/public/MACOSX/marker-chart/',
+        publishedRange: { start: 2000, end: 40000 },
+      }
+  
+      await storeProfileData(exampleProfileData);
+      await storeProfileData(exampleProfileData);
+      await storeProfileData(exampleProfileData);
+  
+      const { getAllByText } = setup();
+      expect(await getAllByText(/Example Profile/)).toHaveLength(3);
```

Comment:
you need to use `findAllByText` instead of `getAllByText` everywhere in this test.

The reason is that the rendering for this component is asynchronous, because of the use of IndexedDB. Maybe we'll change that eventually, by preloading the IDB data, but that's how it is now.
But `getAllByText` isn't asynchronous: this will look up in the DOM __right now__. In contrast `findAllByText` will retry several times (I think 500ms, but I'm not sure) until it finds something, so in this case this will wait until IDB is loaded and the component is fully rendered.